File: csharp/matrix/Matrix.cs
using System;
using System.Linq;

public class Matrix
{
    private readonly int[][] matrix;

    public Matrix(string input)
    {
        matrix = input
            .Split('\n')
            .Select(row => row
                .Split(' ')
                .Select(x => int.Parse(x))
                .ToArray())
            .ToArray();
    }

    public int Rows { get => matrix[0].Length; }

    public int Cols { get => matrix.Length; }

    public int[] Row(int row) => matrix[row];

    public int[] Column(int col) => matrix.Select(row => row[col]).ToArray();
}

File: fsharp/matrix/Matrix.fs
module Matrix

let parse (matrix : string) =
    matrix.Split '\n'
    |> Array.map (fun row -> row.Split ' ' |> Array.map int)

let row index matrix =
    (parse matrix).[index] |> Array.toList

let column index matrix =
    [ for row in parse matrix -> row.[index] ]
